DOC Prepare for docs site (#339)
colymba/GridFieldBulkEditingTools: docs/en/02_bulk-uploader.md

---
title: Bulk Uploader
summary: Upload multiple images or files at once to populate DataObjects managed by a GridField
icon: upload
---

# Bulk uploader

A component for uploading images and/or files in bulk into [`DataObject`](api:SilverStripe\ORM\DataObject) managed by the [`GridField`](api:SilverStripe\Forms\GridField\GridField).

## Usage 1

Simplest usage, add the component to your [`GridFieldConfig`](api:SilverStripe\Forms\GridField\GridFieldConfig) as below. The component will find the first [`Image`](api:SilverStripe\Assets\Image) or [`File`](api:SilverStripe\Assets\File) has_one relation to use on the managed `DataObject`.

```php
use Colymba\BulkUpload\BulkUploader;

$config->addComponent(new BulkUploader());
```

## Usage 2

You can specify which `Image` or `File` field to use and a specific `DataObject` class name to use.

> - `$fileRelationName (string, optional)`: The name of the `Image` or `File` has_one field to use (If your relation is set has 'MyImage' => 'Image', the parameter should be 'MyImage')

Reviewer: ```suggestion
- `$fileRelationName (string, optional)`: The name of the `Image` or `File` `has_one` field to use (If your relation is set as `'MyImage' => Image::class`, the parameter should be `'MyImage'`)
```

Author: Updated

> - `$recordClassName (string, optional)`: The class name of the `DataObject` to create (Usefull if for example your `GridField` holds `DataObject`s of different classes, like when used with the `GridFieldAddNewMultiClass` component.)

Reviewer: ```suggestion
- `$recordClassName (string, optional)`: The class name of the `DataObject` to create (Useful if for example your `GridField` holds `DataObject`s of different classes, like when used with the `GridFieldAddNewMultiClass` component.)
```

Author: Updated
